src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs

```diff
@@ -232,9 +232,22 @@ public static void UpdateMenuItems(this AToolbar toolbar,
 			if (sortedToolbarItems == null || previousMenuItems == null)
 				return;
 
-			var context = mauiContext.Context;
 			var menu = toolbar.Menu;
 
+			// menu items can be deleted by Android after switching activities, removing outdated menu items first
+			if (menu != null)
+			{
+				for (var j = previousMenuItems.Count - 1; j >= 0; j--)
+				{
+					var previousMenuItem = previousMenuItems[j];
+					if (menu.FindItem(previousMenuItem.ItemId) == null)
+					{
+						previousMenuItem.Dispose();
+						previousMenuItems.Remove(previousMenuItem);
```

Comment:
`previousMenuItems.RemoveAt(j);`

Use `RemoveAt(j)` instead of `Remove(previousMenuItem)`.  Remove() which has to search the collection for the item while RemoveAt(j) directly removes at the index we already know, eliminating the search operation.